themes/powerbash10k/powerbash10k.theme.sh

```diff
@@ -0,0 +1,367 @@
+}
+
+# cli
+function __pb10k_show {
+    local seg
+    seg="${1:-}"
+    shift
+    export THEME_SHOW_"$seg"=true
+}
+
+function __pb10k_hide {
+    local seg
+    seg="${1:-}"
+    shift
+    export THEME_SHOW_"$seg"=false
+}
+
+function __pb10k_completion {
+    local cur action actions segments
+    COMPREPLY=()
+    cur="${COMP_WORDS[COMP_CWORD]}"
+    action="${COMP_WORDS[1]}"
+    actions="show hide"
+    segments="battery clock exitcode python ruby scm sudo todo"
+    case "$action" in
+        show)
+        COMPREPLY=( "$(compgen -W "$segments" -- "$cur")" )
+        return 0
+        ;;
+        hide)
+        COMPREPLY=( "$(compgen -W "$segments" -- "$cur")" )
+        return 0
+        ;;
+    esac
+
+    COMPREPLY=( "$(compgen -W "$actions" -- "$cur")" )
+    return 0
+}
+
+function pb10k {
+    local action segments
+    action="${1:-}"
+    shift
+    segments="${*:-}"
+    typeset func
+    case "$action" in
+        show)
+        func=__pb10k_show;;
+        hide)
+        func=__pb10k_hide;;
+    esac
+    IFS=" " read -ra segments <<< "$segments"
+    for seg in "${segments[@]}"; do
+        seg=$(printf "%s" "$seg" | tr '[:lower:]' '[:upper:]')
+        "$func" "$seg"
+    done
+}
+
+complete -F __pb10k_completion pb10k
+
+# Variables
+export SCM_THEME_PROMPT_PREFIX=""
+export SCM_THEME_PROMPT_SUFFIX=""
+
+export RBENV_THEME_PROMPT_PREFIX=""
+export RBENV_THEME_PROMPT_SUFFIX=""
+export RBFU_THEME_PROMPT_PREFIX=""
+export RBFU_THEME_PROMPT_SUFFIX=""
+export RVM_THEME_PROMPT_PREFIX=""
+export RVM_THEME_PROMPT_SUFFIX=""
+
+export SCM_THEME_PROMPT_DIRTY=" ${_omb_prompt_bold_brown}✗${_omb_prompt_normal}"
+export SCM_THEME_PROMPT_CLEAN=" ${_omb_prompt_bold_green}✓${_omb_prompt_normal}"
+
+THEME_SHOW_SUDO=${THEME_SHOW_SUDO:-"true"}
+THEME_SHOW_SCM=${THEME_SHOW_SCM:-"true"}
+THEME_SHOW_RUBY=${THEME_SHOW_RUBY:-"false"}
+THEME_SHOW_PYTHON=${THEME_SHOW_PYTHON:-"false"}
+THEME_SHOW_CLOCK=${THEME_SHOW_CLOCK:-"true"}
+THEME_SHOW_TODO=${THEME_SHOW_TODO:-"false"}
+THEME_SHOW_BATTERY=${THEME_SHOW_BATTERY:-"false"}
+THEME_SHOW_EXITCODE=${THEME_SHOW_EXITCODE:-"true"}
+
+THEME_CLOCK_COLOR=${THEME_CLOCK_COLOR:-"$_omb_prompt_bold_white"}
+THEME_CLOCK_FORMAT=${THEME_CLOCK_FORMAT:-"%H:%M:%S"}
+
+__PB10K_PROMPT_CHAR_PS1=${THEME_PROMPT_CHAR_PS1:-"❯"}
+__PB10K_PROMPT_CHAR_PS2=${THEME_PROMPT_CHAR_PS2:-"\\"}
+
+__PB10K_TOP_LEFT=${__PB10K_TOP_LEFT:-"dir scm"}
+__PB10K_TOP_RIGHT=${__PB10K_TOP_RIGHT:-"exitcode cmd_duration user_info python ruby todo clock battery"}
+__PB10K_BOTTOM=${__PB10K_BOTTOM:-"char"}
+__PB10K_PROMPT_LOCAL_USER_INFO=${__PB10K_PROMPT_LOCAL_USER_INFO:-"true"}
+
+# Prompt
+function __pb10k_ps1 {
+    printf "%s%s%s" "$(__pb10k_top)" "$(__pb10k_bottom)" "$_omb_prompt_normal"
+}
+
+function __pb10k_ps2 {
+    local color
+    color=$_omb_prompt_bold_white
+    printf "%s%s%s" "$color" "${__PB10K_PROMPT_CHAR_PS2}  " "$_omb_prompt_normal"
+}
+
+function _omb_theme_PROMPT_COMMAND {
+    exitcode="$?"
```

Comment:
```suggestion
    local exitcode="$?"
```

Note: Bash's variable scoping model is dynamic scoping, i.e., the variables declared (as `local`) at the caller function are visible to the callee functions (in the entire call tree).